camel-idea-plugin/src/main/java/org/apache/camel/idea/service/extension/camel/JavaCamelIdeaUtils.java

```diff
@@ -170,10 +173,14 @@ public boolean acceptForAnnotatorOrInspection(PsiElement element) {
     public PsiClass getBeanClass(PsiElement element) {
         final PsiElement beanPsiElement = getBeanPsiElement(element);
         if (beanPsiElement != null) {
-            PsiElement resolve = beanPsiElement.getReference().resolve();
-            if (resolve instanceof PsiClass) {
-                return (PsiClass) resolve;
+            PsiClass psiClass = null;
+            final PsiElement resolveElement = beanPsiElement.getReference().resolve();
+            if (resolveElement instanceof PsiClass) {
+                psiClass = (PsiClass) resolveElement;
+            } else if (resolveElement instanceof PsiField) {
+                psiClass = ((PsiClassReferenceType) PsiUtil.getTypeByPsiElement(resolveElement)).resolve();
```

Comment:
Can there be a NPE if the getTypeByPsiElement returns null (if it can do so)

Reply:
good point